### PR TITLE
Use annotation rather than label for serving workloads

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

To address https://github.com/kubernetes-sigs/kueue/pull/3803/files#r1879550890

#### Special notes for your reviewer:

No need for release note as this has not been released yet.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mimowo** [comment]
cc @dgrove-oss

File: pkg/util/testingjobs/statefulset/wrappers.go [hunk 1]
```diff
@@ -106,6 +106,15 @@ func (ss *StatefulSetWrapper) PodTemplateSpecLabel(k, v string) *StatefulSetWrap
 	return ss
 }
 
+// PodTemplateAnnotation sets the annotation of the pod template
+func (d *StatefulSetWrapper) PodTemplateAnnotation(k, v string) *StatefulSetWrapper {
```
Comment (mbobrovskyi) [inline pkg/util/testingjobs/statefulset/wrappers.go:110]:
```suggestion
func (ss *StatefulSetWrapper) PodTemplateAnnotation(k, v string) *StatefulSetWrapper {
```

Comment (mimowo) [reply pkg/util/testingjobs/statefulset/wrappers.go]: done

**Review by mbobrovskyi — APPROVED** [review]
/lgtm
Thanks!

File: pkg/controller/jobs/pod/pod_webhook.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ const (
 	ManagedLabelKey              = constants.ManagedByKueueLabel
 	ManagedLabelValue            = "true"
 	PodFinalizer                 = ManagedLabelKey
-	SuspendedByParentLabelKey    = "kueue.x-k8s.io/pod-suspending-parent"
+	SuspendedByParentAnnotation  = "kueue.x-k8s.io/pod-suspending-parent"
```
Comment (tenzen-y) [inline pkg/controller/jobs/pod/pod_webhook.go:52]:
```suggestion
	SuspendedByParentAnnotationKey  = "kueue.x-k8s.io/pod-suspending-parent"
```

**Review by tenzen-y — COMMENTED** [review]
One nit

File: pkg/controller/jobs/pod/pod_webhook.go (hunk 2 above)
Comment (mimowo) [reply pkg/controller/jobs/pod/pod_webhook.go:52]: hm, but other annotation keys don't have the key suffix. So, I propose not to add it for consistency.

Comment (mimowo) [reply pkg/controller/jobs/pod/pod_webhook.go:52]: but let me move down to be grouped with other annotations

Comment (mimowo) [reply pkg/controller/jobs/pod/pod_webhook.go]: moved down to group with other annotations, but I will wait to confirm with adding the Key suffix

**Pull request merged**